### PR TITLE
fix some parameter setting does not work for CPU DDP and bf16 fail in…

### PR DESCRIPTION
**Pull request by sywangyi**
… DDP path

Signed-off-by: Wang, Yi A <yi.a.wang@intel.com>

### CODE REVIEW

**Comment by sywangyi** [comment]
@sgugger @yao-matrix please help review

File: src/accelerate/commands/launch.py [hunk 1]
```diff
@@ -257,6 +257,8 @@ def simple_launcher(args):
 
     current_env = os.environ.copy()
     current_env["USE_CPU"] = str(args.cpu)
+    current_env["MASTER_ADDR"] = args.main_process_ip
+    current_env["MASTER_PORT"] = str(args.main_process_port)
```
Comment (sgugger) [inline src/accelerate/commands/launch.py:261]: Should this be under and `if args.num_machines > 1`? It seems related to multinode training.

**Review by sgugger — COMMENTED** [review]
Thanks for the fixes!

File: src/accelerate/commands/launch.py (hunk 1 above)
Comment (sywangyi) [reply src/accelerate/commands/launch.py]: thanks for the suggestion, a patch is uploaded to fix it

File: src/accelerate/commands/launch.py [hunk 2]
```diff
@@ -257,6 +257,12 @@ def simple_launcher(args):
 
     current_env = os.environ.copy()
     current_env["USE_CPU"] = str(args.cpu)
+    if args.num_machines > 1:
+        current_env["MASTER_ADDR"] = args.main_process_ip
+        current_env["MASTER_PORT"] = str(args.main_process_port)
+    elif args.num_processes > 1:
+        current_env["MASTER_ADDR"] = "127.0.0.1"
+        current_env["MASTER_PORT"] = "29500"
```
Comment (sgugger) [inline src/accelerate/commands/launch.py:265]: If they are needed, I would rather use the ones set par the user than hardcoding a default, which may create hard-to-debug issues for users.

Comment (sywangyi) [reply src/accelerate/commands/launch.py:265]: hi, sgugger, now in the accelerate config, if num_machines == 1, the master ip and port is not set. and if num_process > 1 and DDP is used in one machine, we still need the master ip and port

Comment (sywangyi) [reply src/accelerate/commands/launch.py]: done, thanks for the suggestion

**Comment by sgugger** [comment]
Thansk again for your contribution!

**Pull request merged**